pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/AbstractStructSchema.java

```diff
@@ -80,6 +84,67 @@ public void setSchemaInfoProvider(SchemaInfoProvider schemaInfoProvider) {
         if (reader != null) {
             this.reader.setSchemaInfoProvider(schemaInfoProvider);
         }
+        this.schemaInfoProvider = schemaInfoProvider;
+    }
+
+    public Schema<?> atSchemaVersion(byte[] schemaVersion) {
```

Comment:
I added explicit documentation for edge cases:
- null schemaVersion
- unknown schemaVersion

Also I have enhanced the default implementation, in case of support for multi version this method must be overridden.